packages/react-dom/src/server/ReactPartialRenderer.js

```diff
@@ -686,17 +686,22 @@ class ReactDOMServerRenderer {
         'Unexpected pop.',
       );
     }
-    this.providerStack[this.providerIndex] = null;
```

Comment:
What does `.length` assignment do in V8? I'd like to make sure it doesn't think the array length change is likely to "stay" because it changes all the time. I think just using `null` is more straightforward and less surprising to the engine (even though the typing isn't as nice for us).

Reply:
OK. But, going back to using `null` seems to require testing for `null` when accessing array items by index, since Flow has been told that `null` and `undefined` are expected array item values. Or, is there an alternate Flow syntax that allows us to remove the explicit checks?